packages/govern-console/src/containers/NewProposal/NewProposal.tsx

```diff
@@ -450,8 +453,17 @@ const NewProposal: React.FC<NewProposalProps> = ({ onClickBack, ...props }) => {
         onTransactionFailure: (error) => {
           enqueueSnackbar(error, { variant: 'error' });
         },
-        onTransactionSuccess: () => {},
-        onCompleteAllTransactions: () => {},
+        onTransactionSuccess: (_, receipt: ContractReceipt) => {
+          containerHash = Proposal.getContainerHashFromReceipt(
```

Comment:
@yuetloo  `proposalInstance` instead of `Proposal`.

Reply:
@novaknole , I purposely made getContainerHashFromReceipt() a  static function so that there's no need to instantiate proposal with a provider to use this function as I think it may be useful in the situation where there's no connection to the node but we still want to get the event args.  Everything already parsed by ethers.js, just need to retrieve the value, no need for a signer/provider to get the event args here..

Reply:
Idea is that in the Govern-console, you would need this `getContainerHashFromReceipt` after the transaction has happened. otherwise, what's the point or where do you get receipt from ? So, because of this, provider will always be needed.  `onTransactionSuccess` gets called after executing a transaction so provider anyways is there.. I just said this change, because in the end, I don't want to end up having `Proposal` and `ProposalInstance` instance.  I'd follow your way but we anyways need `provider` here..

Reply:
I agree with you that in govern-console, we only need `getContainerHashFromReceipt` after a successful transaction.  But, getContainerHashFromReceipt() is a function in Govern.js which is designed to be used by anyone. Also, we always need to have Proposal in order to make proposalInstance, I'm not duplicating anything here. It's just a different way to call the function.